stdlib/ext/file-ext.mc

```diff
@@ -105,6 +111,18 @@ utest
   else ("Error reading file","","","")
 with ("Hello", "Next string", "Final", "EOF") in
 
+-- Test reading x amount of characters from the file
+utest
+  match readOpen filename with Some rc then
+    let l1 = match readBytes rc 3 with Some s then s else "" in
+    let l2 = match readBytes rc 4 with Some s then s else "" in
+    let l3 = match readBytes rc 0 with Some s then s else "" in
+    let l4 = match readBytes rc 1 with Some s then s else "" in
+    readClose rc;
+    (l1,l2,l3,l4)
+  else ("Error reading file","","","")
```

Comment:
Maybe just throw an `error` instead? E.g. `else error "..."`

Reply:
It's written this way in the `readLine` function above, but I think this solution is much cleaner. Is there any downside to throwing an `error` in a `utest` though?

Reply:
Well, I feel the utest should fail normally if the tested function behaves incorrectly (i.e. it does not crash but just returns the wrong value).

However, when there is a problem with the test case itself (e.g., the file can not be opened) this should raise an error IMO. This is because this is not a bug in the tested function, but in the test suite itself.

Reply:
I agree!